controllers/helmchart_controller_test.go

```diff
@@ -592,6 +592,41 @@ var _ = Describe("HelmChartReconciler", func() {
 				return now.Status.Artifact.Revision != got.Status.Artifact.Revision &&
 					!storage.ArtifactExist(*got.Status.Artifact)
 			}, timeout, interval).Should(BeTrue())
+
+			When("Setting valid valuesFile attribute", func() {
+				updated := &sourcev1.HelmChart{}
+				Expect(k8sClient.Get(context.Background(), key, updated)).To(Succeed())
+				updated.Spec.ValuesFile = "./testdata/charts/helmchart/override.yaml"
+				Expect(k8sClient.Update(context.Background(), updated)).To(Succeed())
+				got := &sourcev1.HelmChart{}
+				Eventually(func() bool {
+					_ = k8sClient.Get(context.Background(), key, got)
+					return got.Status.Artifact != nil && storage.ArtifactExist(*got.Status.Artifact) &&
+						got.Status.Artifact.Checksum != updated.Status.Artifact.Checksum
+				}, timeout, interval).Should(BeTrue())
+				f, err := os.Stat(storage.LocalPath(*got.Status.Artifact))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(f.Size()).NotTo(BeZero())
+				helmChart, err := loader.Load(storage.LocalPath(*got.Status.Artifact))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(helmChart.Values["testOverride"]).To(BeTrue())
+			})
+
+			When("Setting invalid valuesFile attribute", func() {
+				updated := &sourcev1.HelmChart{}
+				Expect(k8sClient.Get(context.Background(), key, updated)).To(Succeed())
+				chart.Spec.ValuesFile = "invalid.yaml"
+				Expect(k8sClient.Update(context.Background(), updated)).To(Succeed())
+				got := &sourcev1.HelmChart{}
+				Eventually(func() bool {
+					_ = k8sClient.Get(context.Background(), key, got)
+					return got.Status.Artifact != nil && storage.ArtifactExist(*got.Status.Artifact)
+				}, timeout, interval).Should(BeTrue())
+				f, err := os.Stat(storage.LocalPath(*got.Status.Artifact))
+				Expect(err).NotTo(HaveOccurred())
+				Expect(f.Size()).NotTo(BeZero())
+				Expect(got.Status.Artifact.Revision).To(Equal(updated.Status.Artifact.Revision))
```

Comment:
I do not think that this waits for the update to be processed, but instead just confirms the previous state. The best way to confirm the change has actually been processed before confirming state is likely to confirm the `Status.ObservedGeneration` of `got` has changed compared to `updated`.

Reply:
Yes. I was thinking that might be the correct answer for the `valid` tests as well, instead of `Checksum`.